esp-hal/src/gpio/mod.rs

```diff
@@ -769,31 +692,39 @@ impl Io {
     pub fn set_interrupt_priority(&self, prio: Priority) {
         unwrap!(interrupt::enable(Interrupt::GPIO, prio));
     }
-}
 
-impl crate::private::Sealed for Io {}
-
-impl InterruptConfigurable for Io {
-    /// Install the given interrupt handler replacing any previously set
-    /// handler.
-    ///
+    #[cfg_attr(
+        not(multi_core),
+        doc = "Registers an interrupt handler for all GPIO pins."
+    )]
+    #[cfg_attr(
+        multi_core,
+        doc = "Registers an interrupt handler for all GPIO pins on the current core."
+    )]
+    #[doc = ""]
     /// Note that when using interrupt handlers registered by this function,
     /// we clear the interrupt status register for you. This is NOT the case
     /// if you register the interrupt handler directly, by defining a
     /// `#[no_mangle] unsafe extern "C" fn GPIO()` function.
-    fn set_interrupt_handler(&mut self, handler: InterruptHandler) {
+    pub fn set_interrupt_handler(&mut self, handler: InterruptHandler) {
```

Comment:
Huh I was going to make a comment about making all these methods unstable, before realizing none of them are, but the docs somewhat look like `new` is the only unstable method, when it's actually documenting the impl block stability :thinking: 

![image](https://github.com/user-attachments/assets/eb6ca448-400c-4fb6-b9b1-b2fa3064a0dc)

Reply:
https://github.com/esp-rs/esp-hal/issues/3022

Reply:
Yeah this is a bit worse than I wanted.